Cargo.toml

```
@@ -12,22 +12,27 @@ repository = "https://github.com/futursolo/prokio"
 rust-version = "1.60.0"
 
 [dependencies]
-futures = { version = "0.3", default-features = false, features = ["std", "async-await"] }
+futures = { version = "0.3", default-features = false, features = [
+  "std",
+  "async-await",
+] }
 pin-project = "1.0.11"
 pinned = "0.1.0"
+once_cell = "1"
+
+[target.'cfg(all(target_arch = "wasm32", not(target_os = "wasi")))'.dependencies]
+wasm-bindgen-futures = { version = "0.4" }
+gloo = { version = "0.11" }
 
-[target.'cfg(target_arch = "wasm32")'.dependencies]
-wasm-bindgen-futures = "0.4"
-gloo = "0.8"
+[target.'cfg(any(not(target_arch = "wasm32"), target_os = "wasi"))'.dependencies]
+tokio = { version = "1", features = ["rt", "time"] }
```

```suggestion
tokio = { version = "1.21.0", features = ["rt", "time"] }
```

We should limit tokio to a version that has wasi support.

---

> We should limit tokio to a version that has wasi support.

Ummm... I tested it in my project. At least these capabilities are available in the latest version of `tokio`(1.35)...Is 1.21 too old?

---

> This release is the first release of Tokio to intentionally support WASM. The sync,macros,io-util,rt,time features are stabilized on WASM. Additionally the wasm32-wasi target is given unstable support for the net feature.

> https://github.com/tokio-rs/tokio/blob/master/tokio/CHANGELOG.md#1210-september-2-2022

1.21.0 was based on the Changelog of tokio. But I am fine with a newer version as well.

---

> > This release is the first release of Tokio to intentionally support WASM. The sync,macros,io-util,rt,time features are stabilized on WASM. Additionally the wasm32-wasi target is given unstable support for the net feature.
> 
> > https://github.com/tokio-rs/tokio/blob/master/tokio/CHANGELOG.md#1210-september-2-2022
> 
> 1.21.0 was based on the Changelog of tokio. But I am fine with a newer version as well.

Done https://github.com/yewstack/prokio/pull/11/commits/5ba912c82fcbb0db040c43db38ce824e4fa7c7b8